core/src/saros/communication/connection/ConnectionHandler.java

```diff
@@ -177,6 +180,73 @@ public void setCallback(IConnectingFailureCallback callback) {
     this.callback = callback;
   }
 
+  // TODO make private, keep XMPPException in core, move messages to CoreMessages
+  public static String generateHumanReadableErrorMessage(
+      XMPPAccount account, XMPPException xmppException) {
+
+    // as of Smack 3.3.1 this is always null for connection attempts
+    // Throwable cause = e.getWrappedThrowable();
+
+    XMPPError error = xmppException.getXMPPError();
+
+    if (error != null && error.getCode() == 504)
+      return "The XMPP server "
+          + account.getDomain()
+          + " could not be found. Make sure that you are connected to the internet and that you entered the domain part of your JID correctly.\n\n"
+          // TODO re-add when settings are implemented
```

Comment:
Do you mean "when settings are implemented in IntelliJ"?

Reply:
I more or less ignored this message, but added it in a slightly modified way as IntelliJ allows this setting on Account adding and would use it when modified via eclipse.